main.nf

```diff
@@ -1343,6 +1416,98 @@ if (!params.skipAlignment) {
         }
     }
 
+    /*
+    * Step 3+ - Deduplicate bam files based on UMIs
+    */
+    if(params.with_umi) {
+        // preseq does not work on deduplicated BAM file. Pass it the raw BAM file. 
+        bam.into {bam_umitools_dedup; bam_preseq}
```

Comment:
I'm a bit unsure which of the QC tools should be executed on the deduplicated BAM and which on the raw BAM.